File: helpdeskangular/src/app/service/calendar.service.ts
// import { HttpClient } from '@angular/common/http';
// import { Injectable } from '@angular/core';
// import { Observable } from 'rxjs';
// import { environment } from 'src/environments/environment';
// // import { Calendar } from '../entity/Calendar';

// @Injectable({
//   providedIn: 'root'
// })
// export class CalendarService {

//   // 'http://localhost:8080'
//   host = environment.apiUrl;

//   // number of calendars per page(default = 5)
//   pageSize = environment.pageSize;

//   constructor(private http: HttpClient) { }

//   // get calendars by page and based on the search criteria
//   searchCalendars(pageNumber: number, searchTerm: string, status: string): Observable<Calendar[]> {

//     // ex: url = http://localhost:8080/calendar-search?pageNumber=0&pageSize=5&searchTerm=""&status=""
//     return this.http.get<Calendar[]>(
//       `${this.host}/calendar-search?pageNumber=${pageNumber}&pageSize=${this.pageSize}&searchTerm=${searchTerm}&status=${status}`
//     )
//   }

//   // calculate total of calendars for count total pages
//   getTotalOfCalendars(searchTerm: string, status: string): Observable<number> {

//     // ex: http://localhost:8080/total-of-calendars?searchTerm=""&status=""
//     return this.http.get<number>(
//       `${this.host}/total-of-calendars?searchTerm=${searchTerm}&status=${status}`
//     );

//   }

//   // get all calendars
//   getAllCalendars(status: string): Observable<Calendar[]> {

//     // ex: url1 = http://localhost:8080/calendars?status=""
//     // ex: url2= http://localhost:8080/calendars?status="Active"
//     // ex: url3= http://localhost:8080/calendars?status="Inactive"
//     return this.http.get<Calendar[]>(
//       `${this.host}/calendars?status=${status}`
//     )
//   }

//   // create new calendar
//   public createCalendar(calendar: Calendar): Observable<Calendar> {
//     return this.http.post<Calendar>(`${this.host}/calendar-create`, calendar);
//   }

//   // edit existing calendar
//   public editCalendar(calendar: Calendar): Observable<Calendar> {
//     return this.http.put<Calendar>(`${this.host}/calendar-edit`, calendar);
//   }

//   // find calendar by id
//   findById(id: number): Observable<Calendar> {
//     return this.http.get<Calendar>(`${this.host}/calendar-list/${id}`);
//   }

// }

File: helpdeskspringboot/src/main/java/com/ez/service/CalendarService.java
//package com.ez.service;
//
//import com.ez.entity.HDCalendar;
//import com.ez.repository.CalendarRepository;
//import org.slf4j.Logger;
//import org.slf4j.LoggerFactory;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.stereotype.Service;
//
//import javax.persistence.EntityNotFoundException;
//import java.util.List;
//
//import static com.ez.constant.Constant.*;

//@Service
//public class CalendarService {
//
//    private Logger LOGGER = LoggerFactory.getLogger(getClass());
//
//    @Autowired
//    private CalendarRepository calendarRepository;
//
//    // get all calendars.
//    // parameters:
//    //  - status = "": all(active + inactive) calendars
//    //  - status = "Active": all active calendars
//    //  - status = "Inactive": all inactive calendars
//    public List<HDCalendar> getAllCalendars(String status) {
//
//        LOGGER.info("get all calendars");
//
//        return calendarRepository.getAllCalendars(status);
//    }
//
//    // search calendars by pageNumber and based on the search criteria.
//    // parameters:
//    //  - pageNumber: page number
//    //  - pageSize: page size
//    //  - searchTerm: ID, name
//    //  - status: ''(All), 'Active', 'Inactive'
//    public List<HDCalendar> searchCalendars(int pageNumber, int pageSize, String searchTerm, String status) {
//
//        LOGGER.info("search calendars");
//
//        return calendarRepository.searchCalendars(pageNumber, pageSize, searchTerm, status);
//    }
//
//    // calculate total of calendars based on the search criteria
//    public long getTotalOfCalendars(String searchTerm, String status) {
//
//        LOGGER.info("get total of calendars");
//
//        return calendarRepository.getTotalOfCalendars(searchTerm, status);
//    }
//
//    // create a new calendar
//    public HDCalendar createCalendar(HDCalendar calendar) {
//
//        LOGGER.info("create new calendar");
//
//        // new calendar
//        HDCalendar newCalendar = new HDCalendar(calendar.getName(), calendar.getStatus());
//
//        // save new calendar into database
//        calendarRepository.save(newCalendar);
//
//        return newCalendar;
//    }
//
//    // find calendar by calendar id
//    public HDCalendar findById(Long id) throws EntityNotFoundException {
//
//        LOGGER.info("find calendar by id");
//
//        // find calendar by calendar id
//        return calendarRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(NO_CALENDAR_FOUND_BY_ID + id));
//    }
//
//    // update existing calendar
//    public HDCalendar updateCalendar(HDCalendar calendar) throws EntityNotFoundException {
//
//        LOGGER.info("Update calendar");
//
//        // get existing calendar(persistent)
//        HDCalendar existingCalendar = calendarRepository.findById(calendar.getId())
//                .orElseThrow(() -> new EntityNotFoundException(NO_CALENDAR_FOUND_BY_ID + calendar.getId()));
//
//        // set new values to existing calendar
//        existingCalendar.setName(calendar.getName());
//        existingCalendar.setStatus(calendar.getStatus());
//
//        // update existing calendar(persistent)
//        calendarRepository.save(existingCalendar);
//
//        return existingCalendar;
//    }
//
//}
